core/src/main/scala/zio/config/PropertyTree.scala

```diff
@@ -292,10 +296,13 @@ object PropertyTree {
   def unflatten[K, V](map: Map[Vector[K], ::[V]]): List[PropertyTree[K, V]] =
     mergeAll(map.toList.map(tuple => unflatten(tuple._1.toList, tuple._2)))
 
-  def mergeAll[K, V](list: List[PropertyTree[K, V]]): List[PropertyTree[K, V]] =
-    list.foldRight(List[PropertyTree[K, V]](PropertyTree.empty)) {
-      case (tree, acc) => acc.flatMap(tree0 => tree.merge(tree0))
-    }
+  def mergeAll[K, V](list: List[PropertyTree[K, V]]): List[PropertyTree[K, V]] = list.reverse match {
+    case Nil => PropertyTree.empty :: Nil
+    case head :: tail =>
+      tail.foldLeft(List(head)) {
+        case (acc, tree) => acc.flatMap(tree0 => tree.merge(tree0))
```

Comment:
The fact that you rewrote this function indicates to me that `foo merge empty` does yield `foo` anymore. Is that true? Merge with empty should always yield the other side...

Reply:
I was trying to avoid sporadic empty nodes.
It's not clear where is the source for those empty nodes. I'll might try to reimplement this functionality in a separate PR.

Reply:
👍